app/models/reservation.rb

```diff
@@ -189,7 +189,22 @@ def can_cancel?
   end
 
   def can_customer_edit?
-    !canceled? && !complete? && before_lock_window?
+    !canceled? && !complete? && (reserve_start_at_editable? || reserve_end_at_editable?)
+  end
+
+  def reserve_start_at_editable?
+    Time.zone.now < reserve_start_at && before_lock_window?
```

Comment:
I don't think so because if the lock window is 15 minutes before the reservation, then this condition would be false for the 15 minute block before the reservation starts. Right?